docs/csharp/programming-guide/classes-and-structs/destructors.md

```diff
@@ -1,85 +1,83 @@
-  
-- [Implementing a Dispose Method](../../../standard/garbage-collection/implementing-dispose.md)  
-  
-- [using Statement](../../language-reference/keywords/using-statement.md)  
+> [!NOTE]
+> The .NET Framework implementation makes every reasonable effort to call finalizers for all of its objects that have not yet been garbage collected, unless such cleanup has been suppressed (by a call to the library method `GC.SuppressFinalize`, for example).
   
-## Example  
+ If you need to perform cleanup reliably when an application exist, register a handler for the <xref:System.AppDomain.ProcessExit?displayProperty=fullName> event. That handler would ensure <xref:System.IDisposable.Dispose?displayProperty=nameWithType> has been called for all objects that require cleanup before application exit. Because you can't call *Finalize* directly, and you can't guarantee the garbage collector calls all finalizers before exit, you must use `Dispose` to ensure resources are freed.
 
- The following example creates three classes that make a chain of inheritance. The class `First` is the base class, `Second` is derived from `First`, and `Third` is derived from `Second`. All three have finalizers. In `Main`, an instance of the most-derived class is created. When the program runs, notice that the finalizers for the three classes are called automatically, and in order, from the most-derived to the least-derived.  
-  
- [!code-csharp[Destructors#1](snippets/destructors/Program.cs#1)]
+## Using finalizers to release resources
+
+In general, C# does not require as much memory management on the part of the developer as languages that don't target a runtime with garbage collection. This is because the .NET garbage collector implicitly manages the allocation and release of memory for your objects. However, when your application encapsulates unmanaged resources, such as windows, files, and network connections, you should use finalizers to free those resources. When the object is eligible for finalization, the garbage collector runs the `Finalize` method of the object.
+
+## Explicit release of resources
+
+If your application is using an expensive external resource, we also recommend that you provide a way to explicitly release the resource before the garbage collector frees the object. To release the resource, implement a `Dispose` method from the <xref:System.IDisposable> interface that performs the necessary cleanup for the object. This can considerably improve the performance of the application. Even with this explicit control over resources, the finalizer becomes a safeguard to clean up resources if the call to the `Dispose` method fails.
+
+For more information about cleaning up resources, see the following articles:  
+
+- [Cleaning Up Unmanaged Resources](../../../standard/garbage-collection/unmanaged.md)
+- [Implementing a Dispose Method](../../../standard/garbage-collection/implementing-dispose.md)
```

Comment:
We might want to also add _implementing-disposeasync.md_, if this is also applicable to `DisposeAsync`